pdr_backend/pdr_dashboard/util/db.py

```diff
@@ -75,16 +77,13 @@ def _query_db(self, query: str, scalar=False) -> Union[List[dict], Exception]:
             dict: Query result.
         """
         try:
-            db = DuckDBDataStore(self.lake_dir, read_only=True)
-
+            # If scalar, fetch a single result
             if scalar:
-                result = db.query_scalar(query)
-            else:
-                df = db.query_data(query)
-                result = df.to_pandas()
-
-            db.duckdb_conn.close()
-            return result
+                result = duckdb.execute(query).fetchone()
```

Comment:
wait, why did we remove the query_scalar and moved it here?

Reply:
because query_scalar works on an existing db connection within duckdb data store, we don't want to have a new db connection